app/controllers/AddBusinessNameController.scala

```diff
@@ -76,12 +76,15 @@ class AddBusinessNameController @Inject()(authenticate: AuthenticationPredicate,
   def handleRequest(isAgent: Boolean, backUrl: String)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
     Future {
       if (isDisabled(IncomeSources)) {
-        Redirect(controllers.routes.HomeController.show())
+        Redirect(
+          if(isAgent) controllers.routes.HomeController.showAgent
```

Comment:
if you can wrap Redirect into Future.Future.successful which is a blocking call and rendering of the view below into actual Future;